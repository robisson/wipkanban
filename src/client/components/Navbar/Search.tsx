import * as React from "react";
import SearchIcon from '@material-ui/icons/Search';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';

const Search = () => {
  return (
    <Tooltip title="Search cards">
      <IconButton color="inherit" aria-label="Search cards">
        <SearchIcon />
      </IconButton>
    </Tooltip>
  );
};

export default Search;
