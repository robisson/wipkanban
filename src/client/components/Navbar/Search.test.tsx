import React from "react";
import Search from "./Search";
import { createShallow } from "@material-ui/core/test-utils";
import Tooltip from "@material-ui/core/Tooltip";
import { ShallowWrapper } from "enzyme";

describe("Search Component", () => {
  let shallow: any;

  beforeEach(() => {
    shallow = createShallow();
  });

  it("should render a Tooltip", () => {
    let wrapper:ShallowWrapper = shallow(<Search />);
    expect(wrapper.type()).toEqual(Tooltip);
  });

  it("should have a prop title", () => {
    let wrapper:ShallowWrapper = shallow(<Search />);
    expect((wrapper.props() as any).title).toEqual("Search cards");
    expect(wrapper).toMatchSnapshot();
  });
});
