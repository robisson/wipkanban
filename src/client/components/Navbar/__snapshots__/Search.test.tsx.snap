// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should have a prop title 1`] = `
<WithStyles(Tooltip)
  title="Search cards"
>
  <WithStyles(IconButton)
    aria-label="Search cards"
    color="inherit"
  >
    <pure(SearchIcon) />
  </WithStyles(IconButton)>
</WithStyles(Tooltip)>
`;
